### ISSUE TITLE
Template Preview

### ISSUE THREAD

**Issue by krvax** [merchantos/PrintTemplates#26]
Hi, 

I'm wondering if there is a tool  to previsualize Templates?

Currently I have to see an existing sale, but I'd like to see the changes in real time.

Thanks for your help.

**Comment by lukebowerman** [comment]
Currently there is not. The best trick is to find a old sale (via the All Transactions report for instance) and open a old receipt via the "Print Receipt" button. 

Then you can change the template, save and reload the receipt to see your changes.

Obviously if you make changes that you don't want to keep you'll need to revert any changes you made.

**Comment by krvax** [comment]
Thanks!

So, do you know if there is a way to call the view through the API?

2015-05-08 18:23 GMT-05:00 Luke Bowerman <notifications@github.com>:

> Currently there is not. The best trick is to find a old sale (via the All
> Transactions report for instance) and open a old receipt via the "Print
> Receipt" button.
>
> Then you can change the template, save and reload the receipt to see your
> changes.
>
> Obviously if you make changes that you don't want to keep you'll need to
> revert any changes you made.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/merchantos/PrintTemplates/issues/26#issuecomment-100392938>
> .
>



-- 
|< r\/4

**Comment by lukebowerman** [comment]
Yes - it's actually done via one of the DisplayTemplate endpoints: /API/Account/{accountID}/Sale/{saleID}

Details at:
http://cloud-docs.merchantos.com/API/Account/797/DisplayTemplate/Sale.help

**Comment by urbanangler** [comment]
Is it possible for you guys to upload a sample TWIG Fiddle so we can work on it in real time like Krvax suggested? I am looking for a way to alter the print template for items. A WYSIWYG in the system would be ideal but maybe you could post a sample to http://twigfiddle.com/

**Comment by lukebowerman** [comment]
I gave TWIG fiddle a try but unfortunately we use several extensions throughout the templates so our templates won't work against it easily.

**Comment by urbanangler** [comment]
Thanks for trying it out. It is really too bad that nothing exists though…we have a web dev that doesn’t know TWIG and thus are left to tying to figure this out by stabbing in the dark. Unfortunately we can’t leave it as is due to some issues of not being able to read the item name. Who do we call for help? 

I’ll tell you what is going on just in case you are the guy who could help us. The simple solution can’t be done: i.e. change the name. The reason we can’t do that is because these items are listed in our webstore and because everything in the cloud is linked we can’t use that as a solution.

Details:
Print dimensions are: 1.25 x 1 
Current default template with one <br> added in front of description <p> tag and no other changes.




We just want the names of these items to fit on the label but it looks like the blocks are defined in a non-intuitive way so we can’t really seem to adjust them correctly even after trying the customizable label template. Maybe just putting the price under the barcode.

Any suggestions?

Thanks,

Karl Andersen
Urban Angler
381 5th Ave. 2nd Floor
New York, NY 10016
(212) 689-6400
karl@urbanangler.com <mailto:karl@urbanangler.com>
